src/main/java/org/jpeek/calculus/GeneralCalculation.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2019 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.calculus;
+
+import com.amazonaws.util.IOUtils;
+import com.jcabi.log.Logger;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Calculates aggregated value for cohesion metrics.
+ *
+ * @since 1.0.0
+ * @checkstyle StringLiteralsConcatenationCheck (250 lines)
+ */
+public class GeneralCalculation {
+
+    /**
+     * Created report with aggregated value.
+     *
+     * @param target Path to report file
+     * @param metrics List of metrics used in calculation
+     * @throws IOException If fails
+     */
+    @SuppressWarnings({
+        "PMD.AvoidDuplicateLiterals",
+        "PMD.ExcessiveMethodLength"
+    })
+    public void createReport(final Path target, final List<String> metrics) throws IOException {
+        final Map<String, MetricPresentation> res = new HashMap<>();
+        for (final String metric : metrics) {
+            res.put(
+                metric,
+                new MetricPresentation(
+                    String.format(
+                        "%s/%s.html",
+                        target.toString(),
+                        metric
+                    )
+                )
+            );
+        }
+        final Map<String, String[]> output = getResult(res);
+        String template;
+        template = IOUtils.toString(
+            GeneralCalculation.class.getResourceAsStream("general_calculus.html")
+        );
+        final String classesout = String.format(
+            "<p id=\"classes\">Classes: %s.</p>\n",
```

Comment:
@iMaks99 if you are generating an HTML, it's better to make it valid. Currently, it doesn't have a header, doesn't have all necessary parts an HTML document has to have. Maybe it's better to use the mechanism for HTML generation, which we already have in the project, through XSL?

Reply:
The idea behind fix via _Java_ code was to avoid using _XSL_ due to lack of experiense.